calcom/cal.com
# feat: generate transcription from recording and API endpoint

## What does this PR do?


- Fixes https://github.com/calcom/cal.com/issues/15347
- Fixes CAL-3872 (Linear issue number - should be visible at the bottom of the GitHub issue description)

API Response 

![Screenshot 2024-06-06 at 10 42 31 PM](https://github.com/calcom/cal.com/assets/53316345/b83d1be2-64de-4704-9ca9-d34a711a8bad)




## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?
To test this locally 

1) You can comment this code in apps/web/pages/api/recorded-daily-video.ts

```
  // if (req.headers["x-webhook-signature"] !== computed_signature) {
  //   return res.status(403).json({ message: "Signature does not match" });
  // }
```

2) and make a POST request to /api/recorded-daily-video with body

you can book a meeting locally with daily video and start recording and then get the BookingReference uid from prisma studio and recording id can be fetched from /bookings/past (check network tab) and use booking reference UID as <ROOM_NAME> and recording id as <RECORDING_ID> below

```
{
  "version": "1.1.0",
  "type": "recording.ready-to-download",
  "id": "46-rec-rtd-81db0db6-eb96-4021-a1f6-48c813d542d7",
  "payload": {
    "type": "cloud",
    "recording_id": <RECORDING_ID>,
    "end_ts": 1717688214,
    "room_name": <ROOM_NAME>,
    "start_ts": 1717687980,
    "status": "finished",
    "max_participants": 1,
    "duration": 234,
    "s3_key": "fksfs"
  },
  "event_ts": 1717688213.803
}
```

You should get {message : "success" }

3) then you can make a request to https://api.daily.co/v1/batch-processor?recordingId=9c5ace2c-8c16-416c-9123-b4bf90f4e60d to see if transcription job was submitted.

It takes about 5 - 15 minutes till this batch processor job completes and you can access the transcript download link using the endpoint
